addons/account/wizard/base_document_layout.py

```diff
@@ -1,13 +1,72 @@
-from odoo import models
+from odoo import api, Command, fields, models
+from odoo.tools import is_html_empty
 
 
 class BaseDocumentLayout(models.TransientModel):
     _inherit = 'base.document.layout'
 
+    from_invoice = fields.Boolean()
+    qr_code = fields.Boolean(related='company_id.qr_code', readonly=False)
+    vat = fields.Char(related='company_id.vat', readonly=False,)
+    account_number = fields.Char(compute='_compute_account_number', inverse='_inverse_account_number',)
+
     def document_layout_save(self):
         """Save layout and onboarding step progress, return super() result"""
         res = super(BaseDocumentLayout, self).document_layout_save()
         if step := self.env.ref('account.onboarding_onboarding_step_base_document_layout', raise_if_not_found=False):
             for company_id in self.company_id:
                 step.with_company(company_id).action_set_just_done()
+            # When we finish the configuration of the layout, we want the dialog size to be reset to large
+            # which is the default behaviour.
+            if res.get('context'):
+                res['context']['dialog_size'] = 'large'
         return res
+
+    @api.depends('qr_code')
+    def _compute_preview(self):
+        """ This override is needed to add the depends on the qr code and the move """
+        super()._compute_preview()
+        if self.env.context.get('active_model') != 'account.move':
+            return
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
```

Comment:
Not asking for changes, but maybe instead of playing with the context we can add a many2one field linked to the move.